tests/parallel_runner.py

```diff
@@ -0,0 +1,160 @@
+import enum
+import multiprocessing
+import os
+import pickle
+import Queue
+import unittest
+import time
+import traceback
+import sys
+
+
+class ParallelTestSuite(unittest.BaseTestSuite):
+  def __init__(self):
+    super(ParallelTestSuite, self).__init__()
+    self.processes = None
+    self.result_queue = None
+
+  def run(self, result):
+    test_queue = self.create_test_queue()
+    self.init_processes(test_queue)
+    results = self.collect_results()
+    return self.combine_results(result, results)
+
+  def create_test_queue(self):
+    test_queue = multiprocessing.Queue()
+    for test in self.reversed_tests():
+      test_queue.put(test)
+    return test_queue
+
+  def reversed_tests(self):
```

Comment:
Indeed. I tried to write this with a minimum of comments necessary, but that fails to capture the "why" here.
Thanks!